gltfTutorial/gltfTutorial_004_ScenesNodesCamerasAnimations.md

```diff
@@ -0,0 +1,273 @@
+
+```
+traverse(node) {
+    processCamera(node.camera);
+    processMeshes(node.meshes);
+    processOtherElements(node); // Skin, joint, skeleton - discussed later
+    for each (child in node.children) {
+        traverse(child);
+    }
+}
+```
+
+In practice there will some additional information be required for the traversal: The processing of some elements that are attached to nodes will require information about *which* node they are attached to. Additionally, the information about the transforms of the nodes have to be accumulated during the traversal. This will be detailed in the following section.
+
+
+### Local and global transforms
+
+Each node can have a transform. Such a transform will usually define a translation, rotation or a scaling. This transform will be applied to all elements that are attached to the node itself and to all its child nodes. The hierarchy of nodes thus allows to structure the translations, rotations and scalings that are applied to the scene elements.
+
+
+#### Local transforms of nodes
+
+There are different possible representations for the local transform of a node. The transform can be given directly by the `matrix` property of the node. This is an array of 16 floating point numbers that describe the matrix in column-major order:
+
+```javascript
+"node0": {
+    "matrix": [
+        0.0,  1.0,  2.0,  3.0,
+        0.1,  1.1,  2.1,  3.1,
+        0.2,  1.2,  2.2,  3.2,
+        0.3,  1.3,  2.3,  3.3
+    ]
+}    
+```
+
+The matrix defined here is
+
+![Matrix](images/matrix.png)
+
+> **Implementation note:**
+
+> The matrix shown above *seems* to be transposed compared to what is written in the JSON. However, matrices in OpenGL are *always* stored in column-major order. All OpenGL functions (and nearly all OpenGL-related matrix utility libraries) are implemented accordingly, and allow creating the appropriate column-major matrix directly from an array.
+
+The transform of a node can also be given using the `transform`, `rotation` and `scale` properties of a node - which is sometimes abbreviated as *TRS*:  
+
+```javascript
+"node0": {
+    "translation": [ 10.0, 20.0, 30.0 ],
+    "rotation": [ 0.7071, 0.0, 0.0, 0.7071 ],
+    "scale": [ 2.0, 1.0, 0.5 ]
+}
+```
+
+This representation can be imagined as simply performing the transformations, one after the other. The `translation` just contains the translation in x-, y- and z-direction. The `scale` contains the scaling factors along the x-, y- and z-axis. The trickiest part is the `rotation`, which is given as a [quaternion](https://en.wikipedia.org/wiki/Quaternion). The mathematical background of quaternions is beyond the scope of this tutorial. For now, the most important information is that a quaternion is a compact representation of a rotation about an arbitrary angle and around an arbitrary axis.

```

Comment:
The subsequent section explained this in more detail, and emphasized the importance of computing the matrix as M=TRS. (In fact, I thought that https://github.com/KhronosGroup/glTF/commit/eb4afeaae8c3d9e320b84906d10b368cc0f26b3e might have been caused by someone noticing this in the tutorial). 

However, I can try to elaborate the TRS computation, maybe add concrete examples of simple T,R and S matrices, and try to explain that this will eventually be equivalent to scaling, then rotating, and then translating the vertices.